src/console/spcm_control/tx_device.py

```diff
@@ -201,6 +202,49 @@ class attribute is overwritten.
         self.log.debug("Device setup completed")
         self.log_card_status()
 
+    def set_gradient_offsets(self, offsets: Dimensions) -> None:
+        """Set offset values of the gradient output channels.
+
+        Parameters
+        ----------
+        offsets
+            Offset values as Dimension datatype as defined in acquisition parameter
+
+        Returns
+        -------
+            List of integer values read from card for x, y and z offset values
+        """
+        if abs(offsets.x) > self.max_amplitude[1]:
+            self.log.error("Gradient offset of channel x exceeds maximum amplitude.")
+        if abs(offsets.y) > self.max_amplitude[2]:
+            self.log.error("Gradient offset of channel y exceeds maximum amplitude.")
+        if abs(offsets.z) > self.max_amplitude[3]:
+            self.log.error("Gradient offset of channel z exceeds maximum amplitude.")
+
+        # Set offset values
+        spcm.spcm_dwSetParam_i32(self.card, spcm.SPC_OFFS1, offsets.x)
```

Comment:
I can see that we read the values and check that if they are correctly set. Just an opinion, shall we add a redundancy check here as well for the existance of card instance?

Reply:
Good point, I will add the check